Routes/Question/Qsts.js

```diff
@@ -58,135 +69,101 @@ router.post('/', function(req, res) {
    });
 });
 
-router.get('/:cnvId', function(req, res) {
-   req.cnn.chkQry('select id, ownerId, title, UNIX_TIMESTAMP(lastMessage)' +
-       ' * 1000 as lastMessage from Conversation where id = ?', 
-       parseInt(req.params.cnvId),
-       function(err, cnvs) {
-          if (!err && req.validator.check(cnvs.length, Tags.notFound))
-             res.json(cnvs[0]);
-         
-       });
-    req.cnn.release();
-});
-
-router.put('/:cnvId', function(req, res) {
-   var vld = req.validator;
+// edit specific question, must be owner or Admin
+router.put('/:qstId', function(req, res) {
    var body = req.body;
    var cnn = req.cnn;
-   var cnvId = req.params.cnvId;
-
-   async.waterfall([
-   function(cb) {
-      cnn.chkQry('select * from Conversation where id = ?', [cnvId], cb);
-   },
-   function(cnvs, fields, cb) {
-      if (vld.check(cnvs.length, Tags.notFound, null, cb) &&
-       vld.checkPrsOK(cnvs[0].ownerId, cb) && (!body.title ||
-       vld.check(body.title.length < 80, Tags.badValue, ["title"], cb)))
-         cnn.chkQry('select * from Conversation where id <> ? && title = ?',
-          [cnvId, body.title], cb);
-   },
-   function(sameTtl, fields, cb) {
-      if (vld.check(!sameTtl.length, Tags.dupTitle, null, cb))
-         cnn.chkQry("update Conversation set title = ? where id = ?",
-          [body.title, cnvId], cb);
-   }],
-   function(err) {
-      if (!err)
-         res.status(200).end();
-      req.cnn.release();
-   });
-});
-
-router.delete('/:cnvId', function(req, res) {
    var vld = req.validator;
-   var cnvId = req.params.cnvId;
-   var cnn = req.cnn;
+   var qstId = req.params.qstId;
+   var titleLimit = 500, answerLimit = 100;
 
    async.waterfall([
    function(cb) {
-      cnn.chkQry('select * from Conversation where id = ?', [cnvId], cb);
+      cnn.chkQry('select * from Question where id = ?', qstId,
+       function(err, qst) {
+         if(!err && vld.check(qst.length, Tags.notFound)) {
+            cb(qst[0]);
+         }
+       });
    },
-   function(cnvs, fields, cb) {
-      if (vld.check(cnvs.length, Tags.notFound, null, cb) &&
-       vld.checkPrsOK(cnvs[0].ownerId, cb))
-         cnn.chkQry('delete from Conversation where id = ?', [cnvId], cb);
+   function(qst, cb) {
+      if (vld.checkPrsOK(qst.ownerId, cb) &&
+       vld.check(body.categoryId, null, null, cb)) {
```

Comment:
What's this check for? I think it would be better to use vld.hasField, or put an error tag